dcos/nodeutil/util.go

```diff
@@ -364,6 +377,36 @@ func (d *dcosInfo) state(ctx context.Context) (state State, err error) {
 	return state, err
 }
 
+// TaskCanonicalID return a CanonicalID for a given task.
+func (d *dcosInfo) TaskCanonicalID(ctx context.Context, task string) (*CanonicalID, error) {
+	state, err := d.state(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, framework := range state.Frameworks {
+		for _, t := range framework.Tasks {
+			if t.Name != task && !strings.Contains(t.ID, task) {
```

Comment:
Is it possible that there could be two tasks with the same name?  If so, it seems like we might match this framework.Task incorrectly here.

Reply:
very good point